main/rfc1867.c

```diff
@@ -1149,9 +1149,39 @@ SAPI_API SAPI_POST_HANDLER_FUNC(rfc1867_post_handler) /* {{{ */
 				snprintf(lbuf, llen, "%s[name]", param);
 			}
 			register_http_post_files_variable(lbuf, s, &PG(http_globals)[TRACK_VARS_FILES], 0);
-			efree(filename);
 			s = NULL;
 
+			/* Add $foo_fullpath */
+			if (llen < strlen(param) + MAX_SIZE_OF_INDEX + 1) {
+				llen = (int)strlen(param);
+				lbuf = (char *) safe_erealloc(lbuf, llen, 1, MAX_SIZE_OF_INDEX + 1);
+				llen += MAX_SIZE_OF_INDEX + 1;
+			}
+
+			if (is_arr_upload) {
+				if (abuf) efree(abuf);
+				abuf = estrndup(param, strlen(param)-array_len);
+				snprintf(lbuf, llen, "%s_fullpath[%s]", abuf, array_index);
+			} else {
+				snprintf(lbuf, llen, "%s_fullpath", param);
+			}
```

Comment:
@nikic Should I remove the whole block or just the else? In 61ba79df17867715b265de8b47238f96e62edf9c you left that in for `$foo_name`.

Reply:
You should drop the whole block, but you may need to move the abuf reallocation into the block below.

Reply:
I removed the whole block altogether, including the abuf. It should already be handled by the `$foo_name` part above.

Thanks for your help, I'm more of a PHP and not a C developer.